drivers/console/aarch64/console.S

```diff
@@ -95,10 +95,15 @@ endfunc console_getc
 	 * data that hasn't been output. It returns 0
 	 * upon successful completion, otherwise it
 	 * returns -1.
-	 * Clobber list : x0, x1
+	 * Clobber list : x0, x1, x2, x3, x4
 	 * ---------------------------------------------
 	 */
 func console_flush
+#if COREBOOT
+	mov	x4, x30
+	bl	coreboot_cbmc_flush
+	mov	x30, x4
+#endif
```

Comment:
I'm struggling to visualise what you mean here. I like the idea of introducing a `console_register()` function, which supersedes use of `console_init()`, while re-implementing the latter in terms of the former to keep compatibility. However, I don't understand what is achieved by your series of assembler macros and duplicate weak functions, or why that is better than what I suggested. Perhaps we just need to see some prototype code?